src/Builder/Checker/ClassMethodChecker.php

```diff
@@ -59,4 +108,102 @@ public function isMethodReturnArray(ClassMethod $node): bool
 
         return false;
     }
+
+    /**
+     * @param \PhpParser\Node\Stmt\ClassMethod|null $methodNode
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isMethodNodeSameAsValue(?ClassMethod $methodNode, $value): bool
+    {
+        if (!$value) {
+            return true;
+        }
+        if (!$methodNode || !$methodNode->stmts) {
+            return true;
+        }
+        $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        $previousValue = $parser->parse('<?php ' . $value);
+        if (!$previousValue) {
+            return false;
+        }
+        if (count($previousValue) != count($methodNode->stmts)) {
+            return false;
+        }
+        if (!$this->isSameMethodsBody($previousValue, $methodNode->stmts)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param mixed $previousValue
+     * @param mixed $currentValue
+     *
+     * @return bool
+     */
+    public function isSameMethodsBody($previousValue, $currentValue): bool
+    {
+        if (is_array($previousValue) && is_array($currentValue)) {
+            foreach ($previousValue as $keyPreviousValue => $valuePreviousValue) {
+                if (!isset($currentValue[$keyPreviousValue])) {
+                    return false;
+                }
+                if (!$this->isSameMethodsBody($valuePreviousValue, $currentValue[$keyPreviousValue])) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        foreach ($this->methodStatementCheckers as $methodStatementChecker) {
+            if ($methodStatementChecker->isApplicable($previousValue, $currentValue)) {
+                return $methodStatementChecker->isSameStatements($previousValue, $currentValue);
+            }
+        }
+
+        return $this->getIsSameForRecursiveFields($previousValue, $currentValue);
+    }
+
+    /**
+     * @param mixed $previousValue
+     * @param mixed $currentValue
+     *
+     * @return bool
+     */
+    protected function getIsSameForRecursiveFields($previousValue, $currentValue): bool
+    {
+        $isSame = true;
+        if ($this->isExistsStatementsField($previousValue, $currentValue, static::METHOD_FIELD_ARGS)) {
+            foreach ($previousValue->args as $keyArgument => $argument) {
+                if (!$isSame) {
+                    break;
+                }
+                $isSame = $this->isSameMethodsBody($argument->value, $currentValue->args[$keyArgument]->value);
```

Comment:
is it possible to refactor with less recursion usage? It's hard to read and very possible to fail at some point

Reply:
anyway we must use recoursion because structure of expressions may be different